### PR TITLE
Support importing primitive list from JSON

### LINKED ISSUE #5362: Array of primitives: Support JSON methods
Split from https://github.com/realm/realm-java/pull/5031

All our `realm.create*FromJSON()` methods should support arrays of primitives.

### PR DESCRIPTION
**Pull request by beeender**
Close #5362

### CODE REVIEW

File: realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java [hunk 1]
```diff
@@ -55,7 +55,7 @@
             "io.realm.internal.OsSchemaInfo",
             "io.realm.internal.OsObjectSchemaInfo",
             "io.realm.internal.Property",
-            "io.realm.internal.ProxyUtils",
+            "io.realm.ProxyUtils",
```
Comment (Zhuinden) [inline realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java]: :o

File: realm/realm-library/src/androidTest/java/io/realm/entities/PrimitiveListTypes.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    public static final String FIELD_DOUBLE_LIST = "doubleList";
+    public static final String FIELD_FLOAT_LIST = "floatList";
+    public static final String FIELD_DATE_LIST = "dateList";
+    public static final String FIELD_BYTE_LIST = "byteList";
+    public static final String FIELD_SHORT_LIST = "shortList";
+    public static final String FIELD_INT_LIST = "intList";
+    public static final String FIELD_LONG_LIST = "longList";
+
+    @SuppressWarnings("unused")
+    private RealmList<String> stringList;
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/entities/PrimitiveListTypes.java]: Should we also test `@Required RealmList<String> nonNullStringList` ?

File: realm/realm-library/src/main/java/io/realm/ProxyUtils.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    static <E> void setRealmListWithJsonObject(
+            RealmList<E> realmList, JSONObject jsonObject, String fieldName) throws JSONException {
+        if (!jsonObject.has(fieldName))  {
+            return;
+        }
+
+        OsList osList = realmList.getOsList();
+        if (jsonObject.isNull(fieldName)) {
+            osList.removeAll();
+        }
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/ProxyUtils.java]: Return here as well to increase readability?

File: realm/realm-library/src/main/java/io/realm/ProxyUtils.java [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+        if (realmList.clazz == Boolean.class) {
+            for (int i = 0; i < arraySize; i++) {
+                if (jsonArray.isNull(i)) {
+                    osList.addNull();
+                } else {
+                    osList.addBoolean(jsonArray.getBoolean(i));
+                }
+            }
+        } else if (realmList.clazz == Float.class) {
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/ProxyUtils.java]: Collapse this with below? 
`if (realmList.clazz == Double.class || realmList.claszz == Float.class)`

File: realm/realm-library/src/main/java/io/realm/ProxyUtils.java [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+        if (elementClass == Boolean.class) {
+            while (jsonReader.hasNext()) {
+                if (jsonReader.peek() == JsonToken.NULL) {
+                    jsonReader.skipValue();
+                    realmList.add(null);
+                } else {
+                    realmList.add(jsonReader.nextBoolean());
+                }
+            }
+        } else if (elementClass == Float.class) {
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/ProxyUtils.java]: Fold into the double case as above?

File: realm/realm-library/src/main/java/io/realm/ProxyUtils.java [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+                if (token == JsonToken.NULL) {
+                    jsonReader.skipValue();
+                    realmList.add(null);
+                } else if (token == JsonToken.NUMBER) {
+                    realmList.add(new Date(jsonReader.nextLong()));
+                } else {
+                    realmList.add(JsonUtils.stringToDate(jsonReader.nextString()));
+                }
+            }
+        } else if (elementClass == Long.class) {
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/ProxyUtils.java]: Fold all these integers into one case?

File: realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java (hunk 1 above)
Comment (cmelchior) [reply realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java]: It is package private, so the class doesn't leak into the public API

File: realm/realm-library/src/main/java/io/realm/ProxyUtils.java (hunk 4 above)
Comment (beeender) [reply realm/realm-library/src/main/java/io/realm/ProxyUtils.java]: `osList.addFloat((float) jsonArray.getDouble(i));` it needs to be casted to float

File: realm/realm-library/src/main/java/io/realm/ProxyUtils.java (hunk 5 above)
Comment (beeender) [reply realm/realm-library/src/main/java/io/realm/ProxyUtils.java]: same, needs cast

File: realm/realm-library/src/main/java/io/realm/ProxyUtils.java (hunk 6 above)
Comment (beeender) [reply realm/realm-library/src/main/java/io/realm/ProxyUtils.java]: same ... needs cast ... otherwise it throws.

File: CHANGELOG.md [hunk 7]
```diff
@@ -70,7 +71,7 @@ The internal file format has been upgraded. Opening an older Realm will upgrade
 ### Enhancements
 
 * [ObjectServer] `SyncUserInfo` now also exposes a users metadata using `SyncUserInfo.getMetadata()`
-* `RealmList` can now contain `String`, `byte[]`, `Boolean`, `Long`, `Integer`, `Short`, `Byte`, `Double`, `Float` and `Date` values. [Queries](https://github.com/realm/realm-java/issues/5361) and [Importing primitive lists from JSON](https://github.com/realm/realm-java/issues/5361) are not supported yet.
+* `RealmList` can now contain `String`, `byte[]`, `Boolean`, `Long`, `Integer`, `Short`, `Byte`, `Double`, `Float` and `Date` values. [Queries](https://github.com/realm/realm-java/issues/5361) and [Importing primitive lists from JSON](https://github.com/realm/realm-java/issues/5362) are not supported yet.
```
Comment (vivekkiran) [inline CHANGELOG.md]: This line can be removed, "[Importing primitive lists from JSON](https://github.com/realm/realm-java/issues/5362) are not supported yet."

Comment (Zhuinden) [reply CHANGELOG.md]: That is the change log for an older version, where it wasn't implemented yet :D

**Pull request merged**